src/Microsoft.TestPlatform.Build/Tasks/TestTaskUtils.cs

```diff
@@ -0,0 +1,222 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.TestPlatform.Build.Tasks;
+
+internal static class TestTaskUtils
+{
+    public static string CreateCommandLineArguments(ITestTask task)
+    {
+        const string codeCoverageString = "Code Coverage";
+        const string vsTestAppName = "vstest.console.dll";
+
+        var isConsoleLoggerSpecifiedByUser = false;
+        var isCollectCodeCoverageEnabled = false;
+        var isRunSettingsEnabled = false;
+
+        var builder = new CommandLineBuilder();
+        builder.AppendSwitch("exec");
+        if (task.VSTestConsolePath != null && !task.VSTestConsolePath.ItemSpec.IsNullOrEmpty())
+        {
+            builder.AppendFileNameIfNotNull(task.VSTestConsolePath);
+        }
+        else
+        {
+            builder.AppendFileNameIfNotNull(vsTestAppName);
+        }
+
+        // TODO log arguments in task
+        if (!task.VSTestSetting.IsNullOrEmpty())
+        {
+            isRunSettingsEnabled = true;
+            builder.AppendSwitchIfNotNull("--settings:", task.VSTestSetting);
+        }
+
+        if (task.VSTestTestAdapterPath != null && task.VSTestTestAdapterPath.Any())
+        {
+            foreach (var arg in task.VSTestTestAdapterPath)
+            {
+                builder.AppendSwitchIfNotNull("--testAdapterPath:", arg);
+            }
+        }
+
+        builder.AppendSwitchIfNotNull("--framework:", task.VSTestFramework);
+
+        // vstest.console only support x86 and x64 for argument platform
+        if (!task.VSTestPlatform.IsNullOrEmpty() && !task.VSTestPlatform.Contains("AnyCPU"))
+        {
+            builder.AppendSwitchIfNotNull("--platform:", task.VSTestPlatform);
+        }
+
+        builder.AppendSwitchIfNotNull("--testCaseFilter:", task.VSTestTestCaseFilter);
+
+        if (task.VSTestLogger != null && task.VSTestLogger.Any())
+        {
+            foreach (var arg in task.VSTestLogger)
+            {
+                builder.AppendSwitchIfNotNull("--logger:", arg);
+
+                if ((arg != null) && arg.StartsWith("console", StringComparison.OrdinalIgnoreCase))
```

Comment:
Nit: un-needed parenthesis

```suggestion
                if (arg != null && arg.StartsWith("console", StringComparison.OrdinalIgnoreCase))
```